lib/bitcoin/BitcoinProcessor.ts

```diff
@@ -176,8 +193,144 @@ export default class BitcoinProcessor {
   }
 
   /**
-   * Trims the databases to the block prior to the block the given transaction is in.
-   * @returns The last fully processed block after trimming. `undefined` if all data are deleted after trimming.
+   * A faster version of process transactions that requires access to bitcoin data directory
+   * @param startingBlock the starting block to begin processing
+   */
+  private async fastProcessTransactions (startingBlock: IBlockInfo) {
+    const bitcoinBlockDataIterator = new BitcoinBlockDataIterator(this.bitcoinDataDirectory!);
+    const bestHeight = await this.bitcoinClient.getCurrentBlockHeight();
+    const bestHash = await this.bitcoinClient.getBlockHash(bestHeight);
+
+    let currentHash = bestHash;
+    let currentHeight = bestHeight;
+    const numOfBlocksToProcess = bestHeight - startingBlock.height + 1;
+
+    const chainInfos: (Map<string, IChainInfo> | undefined)[] = new Array(numOfBlocksToProcess).fill(undefined);
```

Comment:
Might need to change condition in line 256, but is it possible to fill the array with empty maps so:
1. No need to have `undefined` declaration in multiple places.
1. No need for `if` block on line 277.